src/data/VerifierLookupTable.sol

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import {Ownable} from "openzeppelin-contracts/access/Ownable.sol";
+
+import {ITreeVerifier} from "../interfaces/ITreeVerifier.sol";
+
+/// @title Batch Lookup Table
+/// @author Worldcoin
+/// @notice A table that provides the correct tree verifier based on the provided batch size.
+/// @dev It should be used to query the correct verifier before using that verifier for verifying a
+///      tree modification proof.
+contract VerifierLookupTable is Ownable {
+    ////////////////////////////////////////////////////////////////////////////////
+    ///                                   DATA                                   ///
+    ////////////////////////////////////////////////////////////////////////////////
+
+    /// The maximum size of a batch that will be used by the WorldID Identity Manager.
+    uint256 internal constant MAXIMUM_BATCH_SIZE = 1000;
+
+    /// The null address.
+    address internal constant nullAddress = address(0x0);
+
+    /// The null verifier.
+    ITreeVerifier internal constant nullVerifier = ITreeVerifier(nullAddress);
+
+    /// The lookup table for routing batches.
+    ///
+    /// @dev It is one bigger than the batch size so that you can address directly by the batch
+    ///      size.
+    ITreeVerifier[MAXIMUM_BATCH_SIZE + 1] internal verifier_lut;
```

Comment:
This should be a mapping – we're going to have very few of these in practice (like 2 or 3).